tornado-api/src/main/java/uk/ac/manchester/tornado/api/plan/types/WithGridScheduler.java

```diff
@@ -18,16 +18,20 @@
 package uk.ac.manchester.tornado.api.plan.types;
 
 import uk.ac.manchester.tornado.api.ExecutionPlanType;
+import uk.ac.manchester.tornado.api.GridScheduler;
 import uk.ac.manchester.tornado.api.TornadoExecutionPlan;
 
 public final class WithGridScheduler extends ExecutionPlanType {
 
-    public WithGridScheduler(TornadoExecutionPlan parent) {
+    private final GridScheduler gridScheduler;
+
+    public WithGridScheduler(TornadoExecutionPlan parent, GridScheduler gridScheduler) {
         super(parent);
+        this.gridScheduler = gridScheduler;
     }
 
     @Override
     public String toString() {
-        return parentLink.toString() + "\n -> withGridScheduler ";
+        return parentLink.toString() + "\n -> withGridScheduler(<" + gridScheduler + ">)";
```

Comment:
I guess we need `gridScheduler.toString()` right?

Reply:
no need, we override the `toString` method for the class `GridScheduler`